be/src/runtime/routine_load/data_consumer_group.cpp

```diff
@@ -239,22 +239,25 @@ Status KafkaDataConsumerGroup::start_all(StreamLoadContext* ctx) {
             } else {
                 Status st = Status::OK();
                 if (ctx->format == TFileFormatType::FORMAT_AVRO) {
-                    avro_value_t avro;
-                    DeferOp op([&] { avro_value_decref(&avro); });
-                    serdes_schema_t* schema;
-                    serdes_err_t err = serdes_deserialize_avro(serdes, &avro, &schema, msg->payload(), msg->len(),
-                                                               errstr, sizeof(errstr));
-                    if (err) {
-                        LOG(ERROR) << "serdes deserialize avro failed: " << errstr;
-                        return Status::InternalError("serdes deserialize avro failed");
-                    }
-                    char* as_json;
-                    if (avro_value_to_json(&avro, 1, &as_json)) {
-                        LOG(ERROR) << "avro to json failed: %s" << avro_strerror();
-                        return Status::InternalError("avro to json failed");
+                    // We must ensure the msg len > 0.
+                    if (msg->len() > 0) {
+                        avro_value_t avro;
+                        DeferOp op([&] { avro_value_decref(&avro); });
+                        serdes_schema_t* schema;
+                        serdes_err_t err = serdes_deserialize_avro(serdes, &avro, &schema, msg->payload(), msg->len(),
+                                                                   errstr, sizeof(errstr));
+                        if (err) {
+                            LOG(ERROR) << "serdes deserialize avro failed: " << errstr;
+                            return Status::InternalError("serdes deserialize avro failed");
```

Comment:
Return specific error messages as much as possible, we may not be able to see the log in some case such as SAAS.

Reply:
> Return specific error messages as much as possible, we may not be able to see the log in some case such as SAAS.

Updated